ImportXml.py

```diff
@@ -0,0 +1,182 @@
+                                ...
+                                'Year of birth: ': 'A4other'},
+                "subquestions": {
+                    'E7a_SQ001': 'My supervisor is well informed about my field of research.',
+                    'E7a_SQ002': 'My supervisor is available when I need advice.',
+                    'E7a_SQ003': 'My supervisor is open to and respects my research ideas.',
+                    'E7a_SQ004': 'My supervisor gives constructive feedback.',
+                    ...},
+                "section_id": 13901
+            }
+        }
+
+    Raises:
+        ValueError: Invalid XML file
+        ValueError: Provided file has unexpected structure
+    """
+
+    def uniCodeRe(s) -> str:
+        # '\u00a0','\u20ac', '\u00b2', '\u00a7'
+        matched = ["\n", "\t", "\xa0"]
+        # if you want to change the unicode
+        # s_new = s.replace(u'\u2018', '"').replace(u'\u2019', '\'').replace(u'\u00f6', r'ö').replace(u'\u00fc', r'ü')
+        # s_new = s
+        res_s = re.sub("|".join(matched), "", s)
+        return res_s
+
+    def textParse(bsString) -> str:
+        text = ""
+        sps = BeautifulSoup(bsString, "html.parser").find_all("p")
+        # sbs = BeautifulSoup(bsString, 'html.parser').find_all('b')
+        for sp in sps:
+            if bool(sp.string):
+                text += uniCodeRe(sp.string)
+        return text
+
+    def findCategories(resp) -> dict:
+        cates = dict()
+        cats = resp.find_all("category")
+        for cat in cats:
+            cat_lab = cat.label.string
+            cat_val = cat.value.string
+            # A#other case
+            if bool(cat.contingentQuestion):
+                cat_val = cat.contingentQuestion.attrs["varName"]
+                cat_lab = uniCodeRe(cat.contingentQuestion.find("text").get_text())
+            cates[cat_lab] = cat_val
+        return cates
+
+    def findSubquestions(que) -> dict:
+        res_sub = dict()
+        subs = que.find_all("subQuestion")
+        for sub in subs:
+            res_sub[sub["varName"]] = sub.find("text").get_text()
+        return res_sub
+
+    def concatQuesitonText(ques) -> str:
+        res_desc = ""
+        for que in ques:
+            res_desc += uniCodeRe(BeautifulSoup(que.string, "html.parser").get_text())
+        return res_desc
+
+    Ans = dict()
+    try:
+        f = open(filepath, "r")
+    except OSError:
+        print(f"{filepath} is an invalid xml file")
+        sys.exit()
+
+    soup = BeautifulSoup(f, "xml")
+    secs = soup.find_all("section")
+    res_sections = list()
+    res_questions = list()
+    for sec in secs:
+        sec_id = sec.attrs["id"]
+        sec_title = sec.sectionInfo.contents[3].string
+        for info in sec.find_all("sectionInfo"):
+            sec_text = info.find_all("text")
+            text_p = ""
+            for t in sec_text:
+                text_p += textParse(t.string)
+            sec_info = text_p
```

Comment:
Similar here:

```python
section_info = _get_section_info(section)
```

Reply:
ok, I will separate it out.

Reply:
Ok. In this case, I meant by "similar" the separation of the parsing, i.e. similar to `section_title = _get_section_title(section)`. Sorry if I confused you:)

Reply:
ok, got it.